server/src/prompt.rs

```diff
@@ -1,19 +1,40 @@
 // org.freedesktop.Secret.Prompt
 
 use oo7::dbus::ServiceError;
-use zbus::{interface, object_server::SignalEmitter, zvariant::OwnedValue};
+use zbus::{
+    interface,
+    object_server::SignalEmitter,
+    zvariant::{OwnedObjectPath, OwnedValue},
+};
 
-#[derive(Debug)]
-pub struct Prompt {}
+use crate::{prompter_callback::PrompterCallback, Service};
+
+#[derive(Debug, Clone)]
+pub struct Prompt {
+    service: Service,
+    path: OwnedObjectPath,
+}
 
 #[interface(name = "org.freedesktop.Secret.Prompt")]
 impl Prompt {
-    pub async fn prompt(&self, _window_id: &str) -> Result<(), ServiceError> {
-        todo!()
+    pub async fn prompt(
+        &self,
+        _window_id: &str,
```

Comment:
This should use a ashpd::WindowIdentifierType here

Reply:
i know it is not used by the gnome implementation but it will eventually used by something better.